packages/orm/src/test/functional/performance.ts

```diff
@@ -288,12 +382,804 @@ describe("Many-to-many relationship performance", () => {
+      secondaryBusinessTypes: TargetRelationship<NewBusinessType, Relations.ManyToMany>;
+      primaryBusinessType: TargetRelationship<NewBusinessType, Relations.ForeignKey>;
+    };
+    type EmployeeProps = {
+      id: string;
+      appointmentColor: string;
+      calendarTipsReadAt: string;
+      confirmedAt: string;
+      deletedAt: string;
+      email: string;
+      locations: SourceRelationship<typeof Location, Relations.ManyToMany>;
+    };
+    type LocationAddressProps = {
+      id: string;
+      location: SourceRelationship<typeof Location, Relations.OneToOne>;
+    };
+    type NewBusinessTypeProps = {
+      id: string;
+      englishName: string;
+      name: string;
+      pluralName: string;
+      sortingId: number;
+      locationSecondaryBusinessTypes: SourceRelationship<typeof Location, Relations.ManyToMany>;
+      locationPrimaryBusinessType: SourceRelationship<typeof Location, Relations.ForeignKey>;
+    };
+
+    class Room extends Model<typeof Room, RoomProps> implements RoomProps {
+      static modelName = 'Room' as const;
+
+      @Attribute()
+      public id: string;
+
+      @Attribute()
+      public name: string;
+
+      @ForeignKey<Room>('Location', 'rooms')
+      public location: TargetRelationship<Location, Relations.ForeignKey>;
+
+      @ManyToMany<Room>('Service', 'rooms')
+      public services: TargetRelationship<Service, Relations.ManyToMany>;
+    }
+
+    class Service extends Model<typeof Service, ServiceProps> implements ServiceProps {
+      static modelName = 'Service' as const;
+
+      @Attribute()
+      public id: string;
+
+      @Attribute()
+      public name: string;
+
+      @Attribute()
+      public extraTimeInSeconds: string;
+
+      @Attribute()
+      public extraTimeType: string;
+
+      @Attribute()
+      public roomRequired: boolean;
+
+      public rooms: SourceRelationship<typeof Room, Relations.ManyToMany>;
+      public pricingLevels: SourceRelationship<typeof ServicePricingLevel, Relations.ForeignKey>;
+    }
+
+    class ServicePricingLevel extends Model<typeof ServicePricingLevel, ServicePricingLevelProps> implements ServicePricingLevelProps {
+      static modelName = 'ServicePricingLevel' as const;
+
+      @Attribute()
+      public id: string;
+
+      @Attribute()
+      public duration: string;
+
+      @Attribute()
+      public name: string;
+
+      @Attribute()
+      public price: string;
+
+      @Attribute()
+      public deletedAt: string;
+
+      @Attribute()
+      public specialPrice: string;
+
+      @Attribute()
+      public priceType: string;
+
+      @ForeignKey<ServicePricingLevel>('Service', 'pricingLevels')
+      public service: TargetRelationship<Service, Relations.ForeignKey>;
+
+      toObject() {
+        const { service } = this;
+        const { rooms } = service;
+        const serviceRef = service.ref;
+
+        return {
+          ...serviceRef,
+          ...this.ref as any,
```

Comment:
This is something I will need to resolve. Rewriting `ref` getter to a simple function might help here.

Reply:
Solved within https://github.com/surgeventures/noname/pull/13